### PR TITLE
Edge closure

### PR DESCRIPTION
**Pull request by cixzhang**
Okay, this is still broken, but I narrowed down the issue. It seems like ```ll``` fails for my test case. I haven't delved too deeply into ```ll``` yet so that will be my next step. Let me know if you have any ideas what's going on. :D

### CODE REVIEW

**Comment by cixzhang** [comment]
Actually, what might be happening is I'm misunderstanding the format of the return value from ```ll``` and ```ss```. Hm.

**Comment by cixzhang** [comment]
Okay, it is now functioning :D

File: test/hits2.js [hunk 1]
```diff
@@ -70,5 +73,35 @@ describe('hits2', function() {
...
+    var points = [[0,.5],[1,.5],[0,1],[.5,0],[1,1]];
+    var edges = [[0,1],[2,3],[3,4]];
+    var result = hits2.closure(edges, points);
+
+    var expected = {
+      points: [[0,.5],[1,.5],[0,1],[.5,0],[1,1],[.25,.5],[.75,.5]],
+      edges: [[0,5],[5,6],[6,1],[2,5],[5,3],[3,6],[6,4]]
+    };
+
+    function equivalentExists(list, expectedList) {
```
Comment (ptpham) [inline test/hits2.js]: I would actually be okay with a hard equivalent in general. =P

File: lib/hits2.js [hunk 2]
```diff
@@ -66,5 +67,72 @@ function ss(result, x0, d0, l0, x1, d1, l1) {
...
-module.exports = { ll, ss };
+function edgePointsToXDL(x0, x1) {
+  var direction = vec2.create();
+  var length = vec2.distance(x1, x0);
+  vec2.normalize(direction, vec2.sub(_v2_0, x1, x0));
+  return [x0, direction, length];
+}
+
+function dedupe(points) {
+  return _.uniqWith(points, (a, b) => equivalent(a, b));
```
Comment (ptpham) [inline lib/hits2.js]: Yeah! N^2 is totally fine here. If we need more spatial speedup stuff we can add it later. =)

**Pull request merged**